kong/router/atc.lua

```diff
@@ -588,6 +596,119 @@ function _M:exec(ctx)
   return match_t
 end
 
+else  -- is stream subsystem
+
+function _M:select(_, _, _, scheme,
+                   src_ip, src_port,
+                   dst_ip, dst_port,
+                   sni)
+
+  check_select_params(nil, nil, nil, scheme,
+                      src_ip, src_port,
+                      dst_ip, dst_port,
+                      sni)
+
+  local c = context.new(self.schema)
+
+  for _, field in ipairs(self.fields) do
+    if field == "net.protocol" then
+      assert(c:add_value(field, scheme))
+
+    elseif field == "tls.sni" then
+      local res, err = c:add_value(field, sni)
+      if not res then
+        return nil, err
+      end
+
+    elseif field == "net.src.ip" then
+      assert(c:add_value(field, src_ip))
+
+    elseif field == "net.src.port" then
+      assert(c:add_value(field, src_port))
+
+    elseif field == "net.dst.ip" then
+      assert(c:add_value(field, dst_ip))
+
+    elseif field == "net.dst.port" then
+      assert(c:add_value(field, dst_port))
+
+    end -- if
+  end -- for
+
+  local matched = self.router:execute(c)
+  if not matched then
+    return nil
+  end
+
+  local uuid = c:get_result()
+
+  local service = self.services[uuid]
+  local matched_route = self.routes[uuid]
+
+  local service_protocol, _,  --service_type
+        service_host, service_port,
+        service_hostname_type = get_service_info(service)
+
+  return {
+    route          = matched_route,
+    service        = service,
+    upstream_url_t = {
+      type = service_hostname_type,
+      host = service_host,
+      port = service_port,
+    },
+    upstream_scheme = service_protocol,
```

Comment:
`upstream_host` is not appearing here. IMO host means hostname of the upstream, not the HTTP `Host` header. Does hostname makes sense here?

Reply:
`runloop/handler.lua` will pass these info to balancer, see function `preread.before()`